stellargraph/layer/gcn.py

```diff
@@ -0,0 +1,258 @@
+        self.activity_regularizer = regularizers.get(activity_regularizer)
+        self.kernel_constraint = constraints.get(kernel_constraint)
+        self.bias_constraint = constraints.get(bias_constraint)
+        self.support = support
+
+    def compute_output_shape(self, input_shapes):
+        features_shape = input_shapes[0]
+        output_shape = (features_shape[0], self.units)
+        return output_shape  # (batch_size, output_dim)
+
+    def build(self, input_shapes):
+        features_shape = input_shapes[0]
+        assert len(features_shape) == 2
+        input_dim = features_shape[1]
+
+        self.kernel = self.add_weight(
+            shape=(input_dim * self.support, self.units),
+            initializer=self.kernel_initializer,
+            name="kernel",
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+        if self.use_bias:
+            self.bias = self.add_weight(
+                shape=(self.units,),
+                initializer=self.bias_initializer,
+                name="bias",
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+            )
+        else:
+            self.bias = None
+        self.built = True
+
+    def call(self, inputs, mask=None):
+        features = inputs[0]
+        A = inputs[1:]
+
+        supports = list()
+        for i in range(self.support):
+            supports.append(K.dot(A[i], features))
+        supports = K.concatenate(supports, axis=1)
+        output = K.dot(supports, self.kernel)
+
+        if self.bias:
+            output += self.bias
+        return self.activation(output)
+
+    def get_config(self):
+        """
+        Gets class configuration for Keras serialization
+        """
+        config = {
+            "units": self.units,
+            "use_bias": self.use_bias,
+            "support": self.support,
+            "activation": activations.serialize(self.activation),
+            "kernel_initializer": initializers.serialize(self.kernel_initializer),
+            "bias_initializer": initializers.serialize(self.bias_initializer),
+            "kernel_regularizer": regularizers.serialize(self.kernel_regularizer),
+            "bias_regularizer": regularizers.serialize(self.bias_regularizer),
+            "activity_regularizer": regularizers.serialize(self.activity_regularizer),
+            "kernel_constraint": constraints.serialize(self.kernel_constraint),
+            "bias_constraint": constraints.serialize(self.bias_constraint),
+        }
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+
+class GCN:
+    """
+    To create GCN layers with Keras layers.
+
+    """
+
+    def __init__(
+        self,
+        layer_sizes,
+        activations,
+        generator,
+        bias=True,
+        dropout=0.0,
+        normalize=regularizers.l2(5e-4),
```

Comment:
I replaced the default l2 with None but added it in the demo code. Users can fully disable normalization or put their favours' one.

Reply:
Once again, PLEASE fix the meaning of this argument! It's currently not what it is described as in the doc string! Also, if you're going to use it as kernel normalizer, rename it to kernel_normalizer, to avoid shadowing the `normalize` argument of `GraphSAGE` and `HinSAGE` classes, where this is the normalization applied to the output of the last layer of the stack!

Reply:
@wangzhen263 you forgot to update the name from `normalize` to `kernel_regularizer` in the doc string. Please fix.